### PR TITLE
Add benchmarking for `rm`

### PR DESCRIPTION
**Pull request by Benjscho**
Add benchmarking script and guide for `rm`, mostly copied from `ls` benchmarking guide. Tested `rm` using `jwalk` instead of `walkdir`, and saw a slight performance regression, if any change.

### CODE REVIEW

File: src/uu/rm/BENCHMARKING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+```shell
+cargo flamegraph --cmd coreutils -- rm [additional parameters]
+```
+
+However, if the `-r` option is given, the output becomes pretty much useless due to recursion. We can fix this by merging all the direct recursive calls with `uniq`, below is a `bash` script that does this.
+
+```shell
+#!/bin/bash
+cargo build --release --no-default-features --features rm
+perf record target/release/coreutils rm "$@"
```
Comment (sylvestre) [inline src/uu/rm/BENCHMARKING.md:49]: you might want to try https://github.com/mstange/samply

Comment (Benjscho) [reply src/uu/rm/BENCHMARKING.md]: That's interesting, just tried samply and `rm -rf` immediately enters into `std::sys::unix::fs::remove_dir_impl::remove_dir_all_recursive`, which is then 100% of the rest of the samples.

This is likely why `rm -rf` performance isn't improved by `jwalk` at all. We might see improvements if we were manually overriding the `remove_dir_all_recursive`, which appears to be causing [other issues](https://github.com/uutils/coreutils/issues/3722) with GNU compatibility.

Comment (Benjscho) [reply src/uu/rm/BENCHMARKING.md]: This test was again using the linux kernel with `rm -rf`

image

**Comment by Benjscho** [comment]
Only check failing is spelling due to "samply" and "flamegraphs". Is there a way to add words to a spelling ignore dictionary?

**Comment by cakebaker** [comment]
You can add a HTML comment, see for example https://raw.githubusercontent.com/uutils/coreutils/main/CONTRIBUTING.md

**Comment by tertsdiepraam** [comment]
Oh cakebaker was just a bit quicker than me 😄 

I think we could also add these words to the list here: https://github.com/uutils/coreutils/blob/main/.vscode/cspell.dictionaries/shell.wordlist.txt#L79 We might need them in other files too.

**Comment by Benjscho** [comment]
Thank you both! Just added those, hopefully will fix the next run

**Comment by Benjscho** [comment]
Looks like the style and lint changes are failing due to files that aren't changed. This could be related to the release of Rust 1.72 which introduced some new clippy warnings. Would it be fine to merge these changes as they're unrelated?

**Comment by sylvestre** [comment]
thanks

**Pull request merged**